homeassistant/components/plugwise/entity.py

```diff
@@ -22,50 +15,33 @@ class PlugwiseEntity(CoordinatorEntity[PlugwiseData]):
 
     def __init__(
         self,
-        api: Smile,
         coordinator: PlugwiseDataUpdateCoordinator,
-        name: str,
-        dev_id: str,
+        device_id: str,
     ) -> None:
         """Initialise the gateway."""
         super().__init__(coordinator)
+        self._dev_id = device_id
 
-        self._api = api
-        self._name = name
-        self._dev_id = dev_id
-        self._entity_name = self._name
-
-    @property
-    def name(self) -> str | None:
-        """Return the name of the entity, if any."""
-        return self._name
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return the device information."""
-        data = self.coordinator.data.devices[self._dev_id]
-        device_information = DeviceInfo(
-            identifiers={(DOMAIN, self._dev_id)},
-            name=self._entity_name,
-            manufacturer="Plugwise",
-        )
-
+        configuration_url: str | None = None
         if entry := self.coordinator.config_entry:
-            device_information[
-                ATTR_CONFIGURATION_URL
-            ] = f"http://{entry.data[CONF_HOST]}"
-
-        if model := data.get("model"):
-            device_information[ATTR_MODEL] = model
+            configuration_url = f"http://{entry.data[CONF_HOST]}"
+
+        data = self.coordinator.data.devices[device_id]
+        self._attr_device_info = DeviceInfo(
+            configuration_url=configuration_url,
+            identifiers={(DOMAIN, device_id)},
+            manufacturer=data.get("vendor"),
+            model=data.get("model"),
+            name=data.get("name"),
+            sw_version=data.get("fw"),
+        )
 
-        if self._dev_id != self.coordinator.data.gateway["gateway_id"]:
-            device_information[ATTR_VIA_DEVICE] = (
+        if device_id != coordinator.data.gateway["gateway_id"]:
+            self._attr_device_info[ATTR_VIA_DEVICE] = (
                 DOMAIN,
```

Comment:
Please add the name here as: `name=data.get("name")`.

Reply:
Got it! Thanks, will adjust 👍

Reply:
Adjusted in [bbf3e3b](https://github.com/home-assistant/core/pull/66074/commits/bbf3e3bc737f7e1c35a976f278f681a009e76eef)